inst/rmd/qc_report.rmd

```diff
@@ -420,7 +423,83 @@ if(has_processed){
 
 Compromised cells, calculated using `miQC` described above, are filtered and removed from the library prior to downstream analyses. 
 The raw counts from all remaining cells are then normalized prior to selection of highly variable genes and dimensionality reduction. 
-Here we show the UMAP (Uniform Manifold Approximation and Projection) embeddings for each cell, coloring each cell by the total number of genes detected per cell.
+The above plot shows the UMAP (Uniform Manifold Approximation and Projection) embeddings for each cell, coloring each cell by the total number of genes detected per cell.
+The below plots show the same UMAP embeddings, coloring each cell by the gene expression of the gene indicated in the label for each panel. The genes chosen for plotting are the top 12 highly variable genes identified in the library.
+
+```{r message=FALSE}
+if(has_processed){
+  
+  if(has_var_genes){
+    
+    # select top genes to plot 
+    top_genes <- processed_meta$highly_variable_genes %>%
+      head(n=12)
+    
+    # grab expression for top genes from counts
+    var_gene_exp <- logcounts(processed_sce[top_genes,]) %>%
+      as.matrix() %>%
+      t() %>%
+      as.data.frame() %>%
+      tibble::rownames_to_column("barcode")
+    
+    # grab rowdata as data frame to later combine with gene expression data
+    # rowdata contains the mapped gene symbol so can use that to label plots instead of ensembl gene id from rownames 
+    rowdata_df <- rowData(processed_sce) %>%
+      as.data.frame() %>%
+      tibble::rownames_to_column("ensembl_id") %>%
+      dplyr::select(ensembl_id, gene_symbol) %>%
+      dplyr::filter(ensembl_id %in% top_genes)
```

Comment:
I _think_ this will put everything in the correct order.

Note I also changed the pipe operator to match the earlier part of the notebook. This notebook also doesn't use dplyr prefixes in general. We don't load many packages, but I tend to feel that `dplyr` is okay in certain contexts... and I guess I had decided this was one of them.

```suggestion
    rowdata_df <- rowData(processed_sce) |>
      as.data.frame() |>
      tibble::rownames_to_column("ensembl_id") |>
      select(ensembl_id, gene_symbol) |>
      filter(ensembl_id %in% top_genes) |>
      mutate(ensembl_id = factor(ensembl_id, levels = top_genes)) |>
      arrange(ensembl_id) |>
      mutate(gene_symbol = factor(gene_symbol, levels = gene_symbol))
      
```